packages/flutter_tools/lib/src/android/migrations/gradle_java_version_conflict_migration.dart

```diff
@@ -0,0 +1,72 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../../base/file_system.dart';
+import '../../base/project_migrator.dart';
+import '../../base/version.dart';
+import '../../project.dart';
+import '../android_studio.dart';
+
+final Version _androidStudioFlamingo = Version(2022, 2, 0);
+final RegExp _gradleVersionMatch = RegExp(
+  r'\s*distributionUrl=https\\://services\.gradle\.org/distributions/gradle-((?:\d|\.)+)-(?:all|bin)\.zip');
+final Version _lowestSupportedGradleVersion = Version(7, 3, 0);
```

Comment:
```suggestion
final Version _lowestSupportedGradleVersion = Version(7, 3, 0); // minimum for JDK 17
```

(Or maybe work in a link to that compatibility matrix here. But that's the key line from that matrix for what this code is doing.)